File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex10/Exam10.java
package org.example.ex10;

public class Exam10 {

    public Exam10(ExamUser examUser) {

        if (examUser != null && examUser.getName() != null) {

            if (StringUtils.isNotBlank(examUser.getName())) {
                System.out.println(examUser.getName());
            }
        }

    }

    public static void main(String[] args) {
        new Exam10(new ExamUser());
        new Exam10(new ExamUser("abcd"));
    }
}

class StringUtils {

    public static boolean isNotBlank(String value) {
        return !value.isBlank();
    }
}

class ExamUser {
    private String name;

    public ExamUser() {
    }

    public ExamUser(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex10/Exam10.kt
package ex10

/**
 * 확장함수
 */
fun main() {

    val user = ExamUser(name = "abcd")
    exam10(user)
}

fun exam10(examUser: ExamUser?) {

    examUser?.let {

        if (it.name.isNotNullOrBlank()) {
            println(it.name)
        }
    }

    if (examUser.isNotNull() && examUser?.name.isNotNullOrBlank()) {
        println(examUser?.name)
    }

}

data class ExamUser(
    var name: String? = null
)

// 확장 함수
fun String?.isNotNullOrBlank(): Boolean {
    return !this.isNullOrBlank()
}

fun Any?.isNotNull(): Boolean {
    return this != null
}